sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/clustering/algorithms/LocalitySentiveHashing.scala

```diff
@@ -0,0 +1,144 @@
+package net.sansa_stack.template.spark.rdf
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.rdd.RDD
+import org.apache.jena.graph.Triple
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.Dataset
+import org.apache.spark.ml.feature._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.DataTypes
+import org.apache.spark.mllib.linalg.Vector
+import org.graphframes._
+import org.graphframes.GraphFrame
+import org.apache.spark.ml.evaluation.ClusteringEvaluator
+
+/*
+ *
+ * Clustering
+ * @triplesType - List of rdf subjects and their predicates.
+ * @return - cluster of similar subjects.
+ */
+
+class LocalitySensitiveHashing(spark: SparkSession, nTriplesRDD: RDD[Triple], dir_path: String) extends Serializable {
+
+  def run() = {
+    val parsedTriples = getParsedTriples()
+    val extractedEntity = getOnlyPredicates(parsedTriples)
+    val featuredData_Df: DataFrame = vectoriseText(extractedEntity)
+    val (model: MinHashLSHModel, transformedData_Df: DataFrame) = minHashLSH(featuredData_Df)
+    val dataset: Dataset[_] = approxSimilarityJoin(model, transformedData_Df)
+    clusterFormation(dataset, featuredData_Df)
+  }
+
+  def getParsedTriples(): RDD[(String, String, Object)] = {
```

Comment:
um.. any reason why you are parsing triples in such a way? Are you referring to the local-name of the resource? (e.g. https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/rdf/model/Resource.html#getLocalName--) amd lexical name when it is literal? I think there is no need to do such a parsing. We can exclude this transformation (or keep it but do not make it to specific to depent on "/" only as it is quite hard-coded and it may not work on some cases -- but lets hope that there is always a "/" at the end of the URI).

Reply:
yes, I am doing the same thing. So this kind of parsing is required.

Reply:
um, I still think that you can get it via getLocalName or any other build-in Jena methods. Please, give it a try and let me know.

Reply:
Yes, getLocalName worked. Thanks. I have updated the function.